src/main/java/com/amazonaws/kinesisvideo/demoapp/activity/WebRtcActivity.java

```diff
@@ -259,6 +266,47 @@ private boolean isValidClient() {
         return client != null && client.isOpen();
     }
 
+    private void handlePendingIceCandidates(String clientId) {
+        // Add any pending ICE candidates from the queue for the client ID
+        Queue<IceCandidate> pendingIceCandidatesQueueByClientId = pendingIceCandidatesMap.get(clientId);
+        while(!pendingIceCandidatesQueueByClientId.isEmpty()) {
+            final IceCandidate iceCandidate = pendingIceCandidatesQueueByClientId.peek();
+            final boolean addIce = localPeer.addIceCandidate(iceCandidate);
+            Log.d(TAG, "Added ice candidate after SDP exchange " + iceCandidate + " " + (addIce ? "Successfully" : "Failed"));
+          pendingIceCandidatesQueueByClientId.remove();
+        }
+        // After sending pending ICE candidates, the client ID's peer connection need not be tracked
+        pendingIceCandidatesMap.remove(clientId);
+        peerConnectionFoundMap.remove(clientId);
+    }
+
+    private void checkAndAddIceCandidate(Event message, IceCandidate iceCandidate) {
+        // if answer/offer is not received, it means peer connection is not found. Hold the received ICE candidates in the map.
+        if(!peerConnectionFoundMap.containsKey(message.getSenderClientId())) {
+            Log.d(TAG, "SDP exchange is not complete. Ice candidate " + iceCandidate + " + added to pending queue");
+            // Add ICE candidate to a queue
+            peerConnectionFoundMap.put(message.getSenderClientId(), false);
+            Queue<IceCandidate> pendingIceCandidatesQueueByClientId = new LinkedList<>();
```

Comment:
Not sure I understand this one. In case if your client ID doesn't match to any active/existing peer connections, you then should check the hash map of clientID -> ICE queue. If the entry does't exist, you should create an entry which means creating the queue first. Then, add the ICE candidate to the queue. If, however, the entry already exists (there had been some ICE candidates before) then you simply the current ICE candidate to the existing queue.